### PR TITLE
Make func_test with memcheck attached work

### PR DESCRIPTION
**Pull request by ganwell**
memcheck slows down chirp very much, so I found bugs created by different timing. I also learned about SIGPIPE (we have to ignore it).

Sorry for the large pull request, but once I introduced the problems I had to fix them all.

### CODE REVIEW

File: src/connection.c [hunk 1]
```diff
@@ -727,59 +666,29 @@ ch_cn_shutdown(ch_connection_t* conn, int reason)
...
-            if (SSL_shutdown(conn->ssl) < 0) {
-                EC(chirp,
-                   "Could not shutdown SSL connection. ",
-                   "ch_connection_t:%p",
-                   (void*) conn);
-            } else {
-                ch_cn_send_if_pending(conn);
-            }
-        }
+        /* I remove SSL_shutdown, because many other clients don't do it
```
Comment (winged) [inline src/connection.c]: This comment should be reworded.. Looks like a commit message. Future readers won't know the shutdown was once here in the first place, so don't talk about why it was *removed*. I'd remove that comment completely instead.

File: src/func_test.py [hunk 2]
```diff
@@ -73,47 +117,103 @@ def teardown(self):
...
         self.open_messages -= set(ret)
         assert not self.open_messages
+        self.timeout_open = False
 
 # with settings(max_examples=10):
 
 
-TestBuffer = GenBuffer.TestCase
+with settings(deadline=None, timeout=unlimited):
+    TestFunc = GenFunc.TestCase
```
Comment (winged) [inline src/func_test.py]: I'm not sure this has your desired effect. This only assigns a value to TestFunc.. I assume the intent was to have the tests in TestFunc run under the settings context manager?

Comment (ganwell) [reply src/func_test.py]: That is perfectly fine. Tested it.

File: src/connection.c (hunk 1 above)
Comment (ganwell) [reply src/connection.c]: I added a todo to remove that

**Pull request merged**